### ISSUE TITLE
MongoDB stores reminder times as ISODate

### ISSUE THREAD

**Issue by dmutterSF** [dapr/components-contrib#2837]
## Expected Behavior
MongoDB should store reminder times (registeredtime/expirationtime) in a compatible format
## Actual Behavior
MongoDB stores time objects as ISODate

## Steps to Reproduce the Problem
- Create an actor reminder with dapr runtime version 1.10.6
- Try to read actor reminders

## Release Note

RELEASE NOTE: **FIX** Actor Timers/Reminders stored in incompatible format for mongoDB statestore

**Comment by berndverst** [comment]
I reproduced this using Python. Whether we use ISODate or not is an implementation detail - but it needs to be able to serialize / deserialize correctly both ways. Right now that's not happening and I'm getting this error:

> dapr.clients.exceptions.DaprInternalError: ('error invoke actor method: rpc error: code = Internal desc = error invoke actor method: error from actor service: {"message":"error creating actor reminder: error obtaining reminders for actor type DemoActor: could not parse actor reminders: json: cannot unmarshal object into Go struct field .expirationTime of type string","errorCode":"ERR_ACTOR_REMINDER_CREATE"}', 'ERR_ACTOR_INVOKE_METHOD')

**Comment by berndverst** [comment]
I fixed the issues in this PR: https://github.com/dapr/components-contrib/pull/2858

This will be included in Dapr 1.11. We are currently focused on Dapr 1.11 - but may create a 1.10.8 with this fix.

Please note that the `ISODate` type in Mongo DB is correct. The problem was that this was not converted back correctly to the Go native `time.Time` type. Mongo BSON <> Go data type <> JSON conversions are quite tricky.

**Comment by berndverst** [comment]
@dmutterSF an update: You were onto something here. MongoDB did not use to store Actor Reminder information as ISO Date prior to Dapr 1.10.5. We discovered that a change in 1.10.5 (https://github.com/dapr/dapr/pull/6039) caused this behavior.
1.10.7 therefore was not the correct fix after all -- while that change made everything work with ISO Dates, we discovered there was an issue with data explosion handling empty Reminder data. In the end the root cause was https://github.com/dapr/dapr/pull/6039 and we are preparing another change which will come in a new 1.10.X hotfix and Dapr 1.11.1.

Apologies for the inconvenience.
